api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -242,9 +249,9 @@ public static Cohort intersect(Cohort a, Cohort b) {
 	public static Cohort subtract(Cohort a, Cohort b) {
 		Cohort ret = new Cohort();
 		if (a != null) {
-			ret.getMemberIds().addAll(a.getMemberIds());
+			ret.getMembers().addAll(a.getMembers());
 			if (b != null) {
-				ret.getMemberIds().removeAll(b.getMemberIds());
+                ret.getMembers().removeAll(b.getMembers());
```

Comment:
I did consider this and looked up the javadocs. I found this particular excerpt in the [TreeSet](http://docs.oracle.com/javase/7/docs/api/java/util/TreeSet.html) documentation:

> a TreeSet instance performs all element comparisons using its compareTo (or compare) method, so two elements that are deemed equal by this method are, from the standpoint of the set, equal. 

Please advice.